File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
    codes = {
            "C" : "G",
            "G" : "C",
            "T" : "A",
            "A" : "U",
            }
    
    s = list(dna_strand)
    result = ""
    return result.join(list(map(lambda x: codes[x], s)))

File: ruby/rna-transcription/rna_transcription.rb
class Complement
  def self.of_dna(dna)
    result = dna.split("").map { |d| transform_dna(d) }
    result.join("")
  end
end

def transform_dna(dna)
  codes = {
    'G' => 'C',
    'C' => 'G',
    'T' => 'A',
    'A' => 'U'
  }
  
  if codes.has_key? dna
    return codes[dna]
  else
    return ""
  end
end
